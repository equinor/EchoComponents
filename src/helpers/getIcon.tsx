import React from 'react';
import { ReactComponent as Cable } from '../icons/tags/Cable-icon.svg';
import { ReactComponent as CircuitStarter } from '../icons/tags/Circuit-icon.svg';
import { ReactComponent as ElectricalIcon } from '../icons/tags/Electrical-icon.svg';
import { ReactComponent as Function } from '../icons/tags/Function-icon.svg';
import { ReactComponent as Instrument } from '../icons/tags/Instrument-icon.svg';
import { ReactComponent as JunctionBox } from '../icons/tags/JunctionBox-icon.svg';
import { ReactComponent as Line } from '../icons/tags/Line-icon.svg';
import { ReactComponent as MainEquipment } from '../icons/tags/MainEquipment-icon.svg';
import { ReactComponent as ManualValve } from '../icons/tags/ManualValve-icon.svg';
import { ReactComponent as Signal } from '../icons/tags/Signal-icon.svg';
import { ReactComponent as Tag } from '../icons/tags/Tag-icon.svg';
import { ReactComponent as Telecom } from '../icons/tags/Telecom-icon.svg';
/**
 * Method that returns the relevant Icon for the iconType provided
 * Tag icon is the default JSX.Element Icon returned
 * @export
 * @param {('electrical'
 *         | 'main Equipment'
 *         | 'line'
 *         | 'manual Valve'
 *         | 'circuit/starter'
 *         | 'instrument'
 *         | 'cable'
 *         | 'function'
 *         | 'signal'
 *         | 'telecom'
 *         | 'junction box'
 *         | 'administrative'
 *         | string)} iconType The type of icon you want returned
 * @return {*}  {JSX.Element} Relevant icon for the provided iconType
 */
export function getIcon(
    iconType:
        | 'electrical'
        | 'main equipment'
        | 'line'
        | 'manual valve'
        | 'circuit/starter'
        | 'instrument'
        | 'cable'
        | 'function'
        | 'signal'
        | 'telecom'
        | 'junction box'
        | 'administrative'
        | string
): JSX.Element {
    switch (iconType.toLocaleLowerCase()) {
        case 'electrical':
            return <ElectricalIcon aria-label={'electrical'} />;
        case 'main equipment':
            return <MainEquipment aria-label={'main equipment'} />;
        case 'line':
            return <Line aria-label={'line'} />;
        case 'manual valve':
            return <ManualValve aria-label={'manual valve'} />;
        case 'circuit/starter':
            return <CircuitStarter aria-label={'circuit/starter'} />;
        case 'cable':
            return <Cable title={'cable'} />;
        case 'instrument':
            return <Instrument aria-label={'instrument'} />;
        case 'function':
            return <Function aria-label={'function'} />;
        case 'signal':
            return <Signal aria-label={'signal'} />;
        case 'telecom':
            return <Telecom aria-label={'telecom'} />;
        case 'junction box':
            return <JunctionBox aria-label={'junction box'} />;
        case 'administrative':
            return <Tag aria-label={'tag'} />;
        default:
            return <Tag aria-label={'tag'} />;
    }
}
