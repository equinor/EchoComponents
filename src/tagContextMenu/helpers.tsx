import React from 'react';
import { ReactComponent as Cable } from '../icons/tags/Cable-icon.svg';
import { ReactComponent as CircuitStarter } from '../icons/tags/Circuit-icon.svg';
import { ReactComponent as ElectricalIcon } from '../icons/tags/Electrical-icon.svg';
import { ReactComponent as Function } from '../icons/tags/Function-icon.svg';
import { ReactComponent as Instrument } from '../icons/tags/Instrument-icon.svg';
import { ReactComponent as JunctionBox } from '../icons/tags/JunctionBox-icon.svg';
import { ReactComponent as Line } from '../icons/tags/Line-icon.svg';
import { ReactComponent as MainEquipment } from '../icons/tags/MainEquipment-icon.svg';
import { ReactComponent as ManualValve } from '../icons/tags/ManualValve-icon.svg';
import { ReactComponent as Signal } from '../icons/tags/Signal-icon.svg';
import { ReactComponent as Tag } from '../icons/tags/Tag-icon.svg';
import { ReactComponent as Telecom } from '../icons/tags/Telecom-icon.svg';

export function getIcon(
    iconType:
        | 'Electrical'
        | 'Main Equipment'
        | 'Line'
        | 'Manual Valve'
        | 'Circuit/Starter'
        | 'Instrument'
        | 'Cable'
        | 'Function'
        | 'Signal'
        | 'Telecom'
        | 'Junction Box'
        | 'Administrative'
        | string
): JSX.Element {
    switch (iconType) {
        case 'Electrical':
            return <ElectricalIcon />;
        case 'Main Equipment':
            return <MainEquipment />;
        case 'Line':
            return <Line />;
        case 'Manual Valve':
            return <ManualValve />;
        case 'Circuit/Starter':
            return <CircuitStarter />;
        case 'Cable':
            return <Cable />;
        case 'Instrument':
            return <Instrument />;
        case 'Function':
            return <Function />;
        case 'Signal':
            return <Signal />;
        case 'Telecom':
            return <Telecom />;
        case 'Junction Box':
            return <JunctionBox />;
        case 'Administrative':
            return <Tag />;
        default:
            return <Tag />;
    }
}
