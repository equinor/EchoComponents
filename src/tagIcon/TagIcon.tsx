import React from 'react';
import styles from './tagIcon.module.css';

interface TagIconProps {
    icon: JSX.Element;
    legendColor: string;
}
/**
 * Component that will wrap the provided icon with a background color (usually the legend color)
 *
 * @param {TagIconProps} {
 *  icon: Icon to be wrapped
 *  legendColor: background color to apply. Need to be valid css color
 * }
 * @return {*}  {JSX.Element} Wrapped icon with provided color
 */
export const TagIcon: React.FC<TagIconProps> = ({ icon, legendColor }: TagIconProps): JSX.Element => {
    return (
        <div style={{ background: legendColor }} className={`${styles.icon}`}>
            {icon}
        </div>
    );
};

export default TagIcon;
