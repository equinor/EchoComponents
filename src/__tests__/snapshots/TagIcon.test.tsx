import React from 'react';
import renderer from 'react-test-renderer';
import { getIcon } from '../../helpers/getIcon';
import TagIcon from '../../tagIcon/TagIcon';
import TagIconShadowWrapper from '../../tagIcon/TagIconShadow';

it('renders TagIcon correctly', () => {
    const tree = renderer.create(<TagIcon icon={getIcon('Tag')} legendColor={'#FFF'} />).toJSON();
    expect(tree).toMatchSnapshot();
});

it('renders TagIcon with shadow correctly', () => {
    const tree = renderer
        .create(
            <TagIconShadowWrapper>
                <TagIcon icon={getIcon('Tag')} legendColor={'#FFF'} />
            </TagIconShadowWrapper>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});
