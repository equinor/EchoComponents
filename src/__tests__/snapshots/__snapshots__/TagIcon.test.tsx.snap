// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders TagIcon correctly 1`] = `
<div
  className="icon"
  style={
    Object {
      "background": "#FFF",
    }
  }
>
  <svg
    aria-label="tag"
  />
</div>
`;

exports[`renders TagIcon with shadow correctly 1`] = `
<div
  className="shadow"
>
  <div
    className="icon"
    style={
      Object {
        "background": "#FFF",
      }
    }
  >
    <svg
      aria-label="tag"
    />
  </div>
</div>
`;
